src/transformers/utils/versions.py

```diff
@@ -0,0 +1,92 @@
+"""
+Utilities for working with package versions
+"""
+
+import operator
+import re
+import sys
+
+from packaging import version
+
+import pkg_resources
+
+
+ops = {
+    "<": operator.lt,
+    "<=": operator.le,
+    "==": operator.eq,
+    "!=": operator.ne,
+    ">=": operator.ge,
+    ">": operator.gt,
+}
+
+
+def require_version(requirement, hint=None):
+    """
+    Perform a runtime check of the dependency versions, using the exact same syntax used by pip.
+
+    The installed module version comes from the `site-packages` dir via `pkg_resources`.
+
+    Args:
+    - requirement: pip style definition, e.g.,  "tokenizers==0.9.4", "tqdm>=4.27", "numpy"
+    - hint: what suggestion to print in case of requirements not being met
+    """
+
+    # note: while pkg_resources.require_version(requirement) is a much simpler way to do it, it
+    # fails if some of the dependencies of the dependencies are not matching, which is not necessarily
+    # bad, hence the more complicated check - which also should be faster, since it doesn't check
+    # dependencies of dependencies.
+
+    # non-versioned check
+    if re.match(r"^[\w_\-\d]+$", requirement):
+        pkg, op, want_ver = requirement, None, None
+    else:
+        match = re.findall(r"^([^!=<>\s]+)([\s!=<>]{1,2})(.+)", requirement)
+        if not match:
+            raise ValueError(
+                f"requirement needs to be in the pip package format, .e.g., package_a==1.23, or package_b>=1.23, but got {requirement}"
+            )
+        pkg, op, want_ver = match[0]
+        if op not in ops:
+            raise ValueError(f"need one of {list(ops.keys())}, but got {op}")
+
+    # special case
+    if pkg == "python":
+        got_ver = ".".join([str(x) for x in sys.version_info[:3]])
+        if not ops[op](version.parse(got_ver), version.parse(want_ver)):
+            raise pkg_resources.VersionConflict(
+                f"{requirement} is required for a normal functioning of this module, but found {pkg}=={got_ver}."
+            )
+        return
+
+    # check if any version is installed
+    try:
+        got_ver = pkg_resources.get_distribution(pkg).version
+    except pkg_resources.DistributionNotFound:
+        raise pkg_resources.DistributionNotFound(
```

Comment:
I uninstalled `tqdm` and ran 

```
python -c 'import transformers;'
```

I got the following output:

```
pkg_resources.DistributionNotFound: The 'tqdm>=4.27 is required for a normal functioning of this module, but it is not installed.
Try: pip install transformers -U or pip install -e '.[dev]' if you're working with git master' distribution was not found and is required by t, q, d, m, >, =, 4, ., 2, 7
```

=> not sure if this `is required by t, q, d, m, >, =, 4, ., 2, 7` is correct or whether it should rather be `is required by tqdm >= 4.27`

Reply:
Yes, that was half-baked. `DistributionNotFound` for some reason behaves differently than `VersionConflict`:

https://github.com/pypa/pkg_resources/blob/6f81a44010d1266494025647dd1e1f0befa5b26b/pkg_resources/__init__.py#L396

but I found a workaround w/o needing to redefine my own exception class. Hope it's OK for now. Should be easy to recode to our own exception classes down the road should the need arise.

Reply:
Fixed:

```
pkg_resources.VersionConflict: tqdm>=4.27 is required for a normal functioning of this module, but found tqdm==4.16.0.
Try: pip install transformers -U or pip install -e '.[dev]' if you're working with git master
```